doc/developer/design/20240625_source_versioning__table_from_sources.md

```diff
@@ -367,6 +367,78 @@ identifies each collection is updated. The migration would do the following:
     -   Drop the existing top-level collection tied to the source, since this is already
         unused and will no longer be owned by any statement
 
+### System Catalog & Introspection Updates
+
+We currently surface details about both Sources and Tables through various system catalog
+tables in both the `mz_catalog` and `mz_internal` schemas. These expose information
+about the objects themselves as well as statistics and statuses emitted from the active dataflows for each source.
+
+We would make the following updates to provide users and Materialize employees the same
+ability to introspect source-fed tables that they currently have for subsources.
+
+#### [mz_catalog.mz_tables](https://materialize.com/docs/sql/system-catalog/mz_catalog/#mz_tables)
+
+`CREATE TABLE .. FROM SOURCE` statements would be added to this catalog table.
+
+Two new columns would be added:
+
+`data_source_type`: distinguishes these tables
+from 'normal' tables that are written to directly by users. The value of this will
+be one of `'direct'` or `'source'`.
+
+`data_source_id`: a nullable column that contains the `id` of the data source,
```

Comment:
My thought was that we may also have tables who's data comes from other 'data sources' in the future -- e.g. if we supported `CREATE TABLE .. FROM WEBHOOK` or `CREATE TABLE .. FROM FIVETRAN`, etc, such that I didn't want to tie this column directly to our existing concept of sources. If we feel that other stuff isn't likely to happen I can switch it to just `source_id`

Reply:
Ah! Ok. Will respond in the other thread above (https://github.com/MaterializeInc/materialize/pull/27907#discussion_r1723625214) to avoid forking the discussion, since they've converged.